mempool/bench_test.go

```diff
@@ -50,3 +111,34 @@ func BenchmarkCacheRemoveTime(b *testing.B) {
 		cache.Remove(txs[i])
 	}
 }
+
+// BenchmarkClistMempoolGetNextTxBytes-8   	       2	 605761241 ns/op
+func BenchmarkClistMempoolGetNextTxBytes(b *testing.B) {
+	benchmarkMempoolGetNextTxBytes(b, enumclistmempool)
+}
+
+// BenchmarkLLRBMempoolGetNextTxBytes-8   	     324	   3757553 ns/op
+func BenchmarkLLRBMempoolGetNextTxBytes(b *testing.B) {
+	benchmarkMempoolGetNextTxBytes(b, enumllrbmempool)
+}
+
+func benchmarkMempoolGetNextTxBytes(b *testing.B, enum mpEnum) {
+	app := kvstore.NewApplication()
+	cc := proxy.NewLocalClientCreator(app)
+	mempool, cleanup := newMempoolWithAppAndConfig(cc, cfg.ResetTestRoot(fmt.Sprintf("mempool_test_%d", enum)), enum)
+	defer cleanup()
+	size := 100000
+	for i := 0; i < size; i++ {
+		rand, _ := rand.Int(rand.Reader, big.NewInt(100))
+		tx := "k" + strconv.Itoa(i) + "=v" + strconv.Itoa(i) + ",f," + strconv.FormatInt(rand.Int64(), 10)
+		txBytes := []byte(tx)
+		mempool.CheckTx(txBytes, nil, TxInfo{})
+	}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		starter, _ := mempool.GetNextTxBytes(1000, 1, nil)
+		for starter != nil {
+			starter, _ = mempool.GetNextTxBytes(1000, 1, starter)
```

Comment:
can record the returned transactions, and then make sure they have descending priorities

Reply:
Does this sound more like a unit test more than a benchmark?

Reply:
yes, but want to make sure the benchmark code itself is correct.

Reply:
I've checked locally for the descendence by simply print out, does it necessary to do the verification in the code?

Reply:
can do something like in https://github.com/QuarkChain/tendermintx/pull/60/files#diff-88e09aeb3afa2d3eb132abe96ff23d8dR337-R341